### PR TITLE
Extend FormatSpec with a data types

### PR DESCRIPTION
**Pull request by EncodePanda**
Show how different formatters are formatting code with a data type definition

### CODE REVIEW

File: test/functional/FormatSpec.hs [hunk 1]
```diff
@@ -204,4 +210,5 @@ unchangedOrmolu =
   \bar s =  do\n\
   \      x <- return \"hello\"\n\
   \      return \"asdf\"\n\
-  \      \n"
+  \      \n\
```
Comment (fendor) [inline test/functional/FormatSpec.hs]:
```suggestion
  \\n\
```

**Comment by alanz** [comment]
Are the spaces in the input data files originally deliberate?

**Comment by fendor** [comment]
@alanz I dont think so. However, I would like to merge this PR now, if everyone's fine with that.

**Comment by alanz** [comment]
I am happy to merge.

**Pull request merged**